src/state/account/accountService.ts

```diff
@@ -65,44 +82,30 @@ export const fetchProfile = (params, currentUser) => {
     ${userFields}
     ${userPreferences}
   `;
-  return terrasoApi
-    .requestGraphQL(query, { email: currentUser.email })
-    .then(_.get('users.edges[0].node'))
-    .then(user => user || Promise.reject('not_found'))
-    .then(user => ({
-      ..._.omit('preferences', user),
-      preferences: parsePreferences(user),
-    }));
+  const result = await terrasoApi.requestGraphQL<UserQuery>(query, {
+    email: currentUser?.email,
+  });
+
+  const user = result.users.edges.at(0);
+  if (user === undefined) {
+    return Promise.reject('not_found');
+  }
+  return parsePreferences(user.node);
 };
 
 // TODO: this is a temporary solution to get the user's email address,
 // the API should have a account query to get the logged in user data
-export const fetchUser = () => {
-  const query = `
-    query user($email: String!){
-      users(email: $email) {
-        edges {
-          node {
-            ...userFields
-            ...userPreferences
-          }
-        }
-      }
-    }
-    ${userFields}
-    ${userPreferences}
-  `;
-  return terrasoApi
-    .requestGraphQL(query, { email: getUserEmail() })
-    .then(_.get('users.edges[0].node'))
-    .then(user => user || Promise.reject('not_found'))
-    .then(user => ({
-      ..._.omit('preferences', user),
-      preferences: parsePreferences(user),
-    }));
+export const fetchUser = async () => {
+  const email = getUserEmail();
+  return fetchProfile(null, email === undefined ? null : { email });
```

Comment:
it is sort of a Typescript thing, since in Typescript you don't need to worry about forgetting to check for null (the compiler will tell you if you do). but it's also my style preference which i don't feel strongly about so i can make these checks shorter since it seems they also looked strange to paul